exercises/resistor-colors/examples/success-standard/src/ResistorColors.hs

```diff
@@ -0,0 +1,34 @@
+module ResistorColors (value) where
+
+import Data.Char (toUpper)
+
+data Color =
+    Black
+  | Brown
+  | Red
+  | Orange
+  | Yellow
+  | Green
+  | Blue
+  | Violet
+  | Grey
+  | White
+  deriving (Eq, Show, Read)
+
+convert :: Color -> Int
+convert Black = 0
```

Comment:
I notice that it would be *possible* to use http://hackage.haskell.org/package/base-4.12.0.0/docs/Prelude.html#t:Enum here.

I have not yet determined whether it would be a *good idea* to do so.

Reply:
I had the same thought.

What you don't get with `deriving Enum` is an explicit annotation that `Black` is 0, etc.

Making an explicit `instance Enum where ...` has the drawback that you need to specify both `toEnum` and `fromEnum`, even though the exercise only needs the one. [There are ways](https://stackoverflow.com/questions/6000511/better-way-to-define-an-enum-in-haskell) to avoid that, but the first way (using a dynamic lookup table) should be discouraged, and the second way (using `Data` and `Typeable`) is probably a little too complicated to serve as a good example.

So I'm content with having a simple `convert` function.